### ISSUE TITLE
Unable to detach Janus handle

### ISSUE THREAD

**Issue by koefmatt** [Ghamza-Jd/jarust#143]
The [detach method](https://github.com/Ghamza-Jd/jarust/blob/51d3e23156d7672b53ffb0cbe852a71d9a7af85b/jarust/src/jahandle.rs#L163) takes a value instead of a reference, leading to a "Cannot move" error when trying to call it. How is this supposed to be called? Sorry if it's obvious and my Rust-foo is just bad... 😓

**Comment by Ghamza-Jd** [comment]
Hey!
Correct, when calling the detach it moves the handle instead of borrowing it, the idea was to invalidate the handle so we can't use it after calling the handle.

A solution would be to wrap it in an `Option`, that way when calling the `detach` we can set it to none.

```rust
struct Example {
    handle: Option<JaHandle>
}

...

fn example(&mut self) {
    self.handle.take().map(|h| h.detach());
}
```

---

On the other hand, borrowing in the detach method looks reasonable so I guess having a new function that borrows `self` is nice
